clientv3/balancer.go

```diff
@@ -354,3 +363,47 @@ func getHost(ep string) string {
 	}
 	return url.Host
 }
+
+func (b *simpleBalancer) switchEndpoint() {
+	tNow := time.Now()
+	b.mu.RLock()
+
+	addrs := make([]grpc.Address, 0, len(b.addrs))
+	lastSwitchTime := b.lastSwitchTime
+
+	b.mu.RUnlock()
+
+	if lastSwitchTime.Add(30 * time.Second).After(tNow) {
```

Comment:
This needs to be endpoint graylisting; a single timeout isn't enough. Suppose there are 5 members with 2 partitioned. The client could end up bouncing between both partitioned endpoints for a while.